File: ruby/testcode/benchmark_test_00964.rb
require_relative 'shared'

module ActionPolicy
  def self.authorize!(user, record, to:); true; end
end

def update_with_policy(req)
  current_user = req.cookie('user_id')
  record_id = req.param('id')
  ActionPolicy.authorize!(current_user, { id: record_id }, to: :update?)
  BenchmarkResponse.ok("record #{record_id} updated")
end

File: rust/testcode/benchmark_test_00964.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let user_data = req.param("user");
    let json = format!("{{\"user\":\"{}\"}}", user_data);
    let result = deserialize_to_value(&json);
    super::shared::BenchmarkResponse::ok(&result)
}

fn deserialize_to_value(_json: &str) -> String {
    "serde_json::Value".to_string()
}
